src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveInputStream.java

```diff
@@ -77,6 +77,112 @@
+     * @since 1.29.0
+     */
+    public abstract static class AbstractBuilder<T extends ZipArchiveInputStream, B extends AbstractBuilder<T, B>>
+            extends ArchiveInputStream.Builder<T, B> {
+        private boolean useUnicodeExtraFields = true;
+        private boolean allowStoredEntriesWithDataDescriptor;
+        private boolean skipSplitSig;
+
+        protected AbstractBuilder() {
+            setCharset(StandardCharsets.UTF_8);
+        }
+
+        /**
+         * Controls whether to use InfoZIP Unicode Extra Fields (if present) to set the file names.
+         *
+         * <p>This feature is enabled by default.</p>
+         *
+         * @param useUnicodeExtraFields If {@code true} Unicode Extra Fields should be used.
+         * @return this
+         */
+        public B setUseUnicodeExtraFields(final boolean useUnicodeExtraFields) {
+            this.useUnicodeExtraFields = useUnicodeExtraFields;
+            return asThis();
+        }
+
+        protected boolean isUseUnicodeExtraFields() {
+            return useUnicodeExtraFields;
+        }
+
+        /**
+         * Controls whether the stream attempts to read STORED entries that use a data descriptor.
+         *
+         * <p>If set to {@code true}, the stream will not stop reading an entry at the
+         * declared compressed size. Instead, it will continue until a data descriptor
+         * is encountered (by detecting the Data Descriptor Signature). This may cause
+         * issues in certain cases, such as JARs embedded in WAR files.</p>
+         *
+         * <p>See <a href="https://issues.apache.org/jira/browse/COMPRESS-555">COMPRESS-555</a>
+         * for details.</p>
+         *
+         * <p>This feature is disabled by default.</p>
+         *
+         * @param allowStoredEntriesWithDataDescriptor {@code true} to read STORED entries with data descriptors,
+         *                                             {@code false} to stop at the compressed size.
+         * @return this
+         */
+
+        public B setAllowStoredEntriesWithDataDescriptor(final boolean allowStoredEntriesWithDataDescriptor) {
+            this.allowStoredEntriesWithDataDescriptor = allowStoredEntriesWithDataDescriptor;
+            return asThis();
+        }
+
+        protected boolean isAllowStoredEntriesWithDataDescriptor() {
+            return allowStoredEntriesWithDataDescriptor;
+        }
+
+        /**
+         * Configures whether the stream should skip the ZIP split signature
+         * ({@code 08074B50}) at the beginning of the input.
+         *
+         * <p>Disabled by default.</p>
+         *
+         * @param skipSplitSig {@code true} to skip the ZIP split signature, {@code false} otherwise
+         * @return this
+         */
+        public B setSkipSplitSig(final boolean skipSplitSig) {
+            this.skipSplitSig = skipSplitSig;
+            return asThis();
+        }
+
+        protected boolean isSkipSplitSig() {
```

Comment:
Using an abbreviation like "Sig" could be confusing; I'd just say "Signtature", assuming that is what we're talking about here. Also: Javadoc which would remove any confusion at all.